doc/wcmp/wcmp-design.md

```diff
@@ -0,0 +1,605 @@
+
+**In scope:**  
+1. WCMP L3 configuration
+
+**Out of scope:**  
+1. WCMP EVPN Type-5 configuration
+
+## Abbreviations
+
+| Term   | Meaning                                   |
+|:-------|:------------------------------------------|
+| SONiC  | Software for Open Networking in the Cloud |
+| WCMP   | Weighted-Cost Multi-Path                  |
+| ECMP   | Equal-Cost Multi-Path                     |
+| UCMP   | Unequal-Cost Multi-Path                   |
+| EVPN   | Ethernet Virtual Private Network          |
+| FRR    | Free Range Routing                        |
+| BGP    | Border Gateway Protocol                   |
+| IP     | Internet Protocol                         |
+| L3     | Layer 3                                   |
+| OA     | Orchestration Agent                       |
+| DB     | Database                                  |
+| NH     | Next Hop                                  |
+| NHG    | Next Hop Group                            |
+| ToR    | Top-Of-Rack                               |
+| API    | Application Programming Interface         |
+| SAI    | Switch Abstraction Interface              |
+| ASIC   | Application-Specific Integrated Circuit   |
+| SWSS   | Switch State Service                      |
+| CLI    | Сommand-line Interface                    |
+| JSON   | JavaScript Object Notation                |
+| YANG   | Yet Another Next Generation               |
+| PYTEST | Python Testing Framework                  |
+| VS     | Virtual Switch                            |
+| PTF    | Packet Test Framework                     |
+
+## List of figures
+
+[Figure 1: WCMP design](#figure-1-wcmp-design)  
+[Figure 2: WCMP OA design](#figure-2-wcmp-oa-design)  
+[Figure 3: WCMP update flow](#figure-3-wcmp-update-flow)  
+[Figure 4: WCMP show flow](#figure-4-wcmp-show-flow)
+
+## List of tables
+
+[Table 1: Frontend event logging](#table-1-frontend-event-logging)  
+[Table 2: Backend event logging](#table-2-backend-event-logging)
+
+# 1 Introduction
+
+## 1.1 Feature overview
+
+In normal ECMP, the route to a destination has multiple next hops and traffic is expected to be equally distributed  
+across these next hops. In practice, flow-based hashing is used so that all traffic associated with a particular flow  
+uses the same next hop, and by extension, the same path across the network.
+
+Weighted ECMP using BGP link bandwidth introduces support for network-wide UCMP to an IP destination.  
+The unequal cost load balancing is implemented by the forwarding plane based on the weights associated  
+with the next hops of the IP prefix. These weights are computed based on the bandwidths of the corresponding  
+multipaths which are encoded in the BGP link bandwidth extended community. Exchange of an appropriate  
+BGP link bandwidth value for a prefix across the network results in network-wide unequal cost multipathing.
+
+WCMP is applicable in a pure L3 network as well as in a EVPN network.
+
```

Comment:
Sonic WCMP HLD that was discussed earlier - https://github.com/sonic-net/SONiC/commit/078eb697adb58f5e8ae1dabeedd87419069bd91f

HLD was not merged, but the swss changes are merged. https://github.com/sonic-net/sonic-swss/pull/1853

Are we using this implementation?

Reply:
@kperumalbfn yes